src/compiler/checker.ts

```diff
@@ -5255,6 +5297,24 @@ namespace ts {
             return (type.flags & TypeFlags.Tuple) && !!(<TupleType>type).elementTypes;
         }
 
+        function getRegularTypeOfObjectLiteral(type: Type): Type {

```

Comment:
No, the point of the method is to remove "freshness" from the type regardless of the kind of type. I suppose we could call it `getRegularTypeOfType`.